misc/python/materialize/buildkite_insights/util/data_io.py

```diff
@@ -9,9 +8,35 @@
 # the Business Source License, use of this software will be governed
 # by the Apache License, Version 2.0.
 
+import hashlib
 import json
+import os
+import subprocess
+from datetime import datetime, timedelta
 from typing import Any
 
+from materialize import MZ_ROOT
+
+PATH_TO_TEMP_DIR = MZ_ROOT / "temp"
+
+
+def get_file_path(
+    pipeline_slug: str, branch: str | None, build_state: str | None
+) -> str:
+    hash_value = hashlib.sha256(b"{branch}-{build_state}").hexdigest()[:8]
+    return f"{PATH_TO_TEMP_DIR}/builds-{pipeline_slug}-params-{hash_value}.json"
```

Comment:
I could also have included the values directly but the name is not really relevant for humans, would require escaping, and may not be future proof when params change (e.g., list of `build_state` instead of a single value).